### PR TITLE
Added instructions to integrate the multicluster plugin in coredns

### PR DESCRIPTION
**Pull request by npintaux**
Instructions to re-compile coredns in order to enable multicluster discovery. The instructions are using a default Kind cluster as a test platform.

### CODE REVIEW

File: site-src/guides/coredns.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
+# Configuring CoreDNS with the multicluster option
+
+The [CoreDNS multicluster plugin](https://coredns.io/explugins/multicluster/) is necessary to be able to reference multicluster services properly. By default, [CoreDNS](https://coredns.io/) does not include this plugin and it is therefore necessary to recompile CoreDNS to build a container image enabling this fonctionality.
```
Comment (jmhbnz) [inline site-src/guides/coredns.md:3]: Minor typo

```suggestion
The [CoreDNS multicluster plugin](https://coredns.io/explugins/multicluster/) is necessary to be able to reference multicluster services properly. By default, [CoreDNS](https://coredns.io/) does not include this plugin and it is therefore necessary to recompile CoreDNS to build a container image enabling this functionality.
```

File: site-src/guides/coredns.md [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+  dns.trace
+  dns.transfer
+  dns.whoami
+  on
+```
+
+This list is important as we will use it and just add the multicluster plugin to recompile CoreDNS.
+
+## Re-compiling CoreDNS
+The easiest way to recompile CoreDNS and generate a new container image is to use the instructions on the [official CoreDNS source repository](https://github.com/coredns/coredns#compilation-from-source). We specifically recommend the compilation with Docker which includes a read-to-use Go environment.
```
Comment (jmhbnz) [inline site-src/guides/coredns.md:111]: I think you mean ready to use?

```suggestion
The easiest way to recompile CoreDNS and generate a new container image is to use the instructions on the [official CoreDNS source repository](https://github.com/coredns/coredns#compilation-from-source). We specifically recommend the compilation with Docker which includes a ready-to-use Go environment.
```

File: site-src/guides/coredns.md [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+Prior to trigger the compilation, it is necessary to modify the `plugin.cfg` file. First, it is highly recommended to keep the same plugins as the ones found in the previous paragraph.
```
Comment (jmhbnz) [inline site-src/guides/coredns.md:113]: Improve grammar:

```suggestion
Prior to triggering the compilation, it is necessary to modify the `plugin.cfg` file. First, it is highly recommended to keep the same plugins as the ones found in the previous paragraph.
```

**Review by jmhbnz — COMMENTED** [review]
In general I think this is a great first iteration, thanks @npintaux.

I found a few minor issues to tidy up, please see below.

File: site-src/guides/coredns.md [hunk 4]
```diff
@@ -0,0 +1,301 @@
+# Configuring CoreDNS with the multicluster option
+
+The [CoreDNS multicluster plugin](https://coredns.io/explugins/multicluster/) is necessary to be able to reference multicluster services properly. By default, [CoreDNS](https://coredns.io/) does not include this plugin and it is therefore necessary to recompile CoreDNS to build a container image enabling this functionality.
```
Comment (lauralorenz) [inline site-src/guides/coredns.md:3]:
```suggestion
There is a ready made [CoreDNS multicluster plugin](https://coredns.io/explugins/multicluster/) you can use as the DNS component against MCS API implementations using EndpointSlices. By default, [CoreDNS](https://coredns.io/) does not include this plugin and it is therefore necessary to recompile CoreDNS to build a container image enabling this functionality.
```

**Review by lauralorenz — COMMENTED** [review]
One change I'd like to make it more clear when the CoreDNS plugin is necessary / desired

**Comment by lauralorenz** [comment]
commit history still seems a lil whacky but github seems to think itll work 🤷  YOLO

/approve

**Pull request merged**